### PR TITLE
junbon/duration_translation

### PR DESCRIPTION
**Pull request by junbon**

### CODE REVIEW

File: lib/Finance/Contract/Longcode.pm [hunk 1]
```diff
@@ -98,7 +98,19 @@ sub shortcode_to_longcode {
     if ($expiry_type eq 'intraday_fixed_expiry') {
         $when_end = [$date_expiry->datetime . ' GMT'];
     } elsif ($expiry_type eq 'intraday') {
-        $when_end = [Time::Duration::Concise->new(interval => $date_expiry->epoch - $date_start->epoch)->as_string];
+        my $interval_string = Time::Duration::Concise->new(interval => $date_expiry->epoch - $date_start->epoch)->as_string;
+        my @strings;
+        my @durations;
+        foreach my $b ('hour', 'minute', 'second') {
+            $DB::single=1;
```
Comment (binarytom) [inline lib/Finance/Contract/Longcode.pm]: leftover debugging line

File: lib/Finance/Contract/Longcode.pm [hunk 2]
```diff
@@ -98,7 +98,19 @@ sub shortcode_to_longcode {
     if ($expiry_type eq 'intraday_fixed_expiry') {
         $when_end = [$date_expiry->datetime . ' GMT'];
     } elsif ($expiry_type eq 'intraday') {
-        $when_end = [Time::Duration::Concise->new(interval => $date_expiry->epoch - $date_start->epoch)->as_string];
+        my $interval_string = Time::Duration::Concise->new(interval => $date_expiry->epoch - $date_start->epoch)->as_string;
+        my @strings;
+        my @durations;
+        foreach my $b ('hour', 'minute', 'second') {
```
Comment (binarytom) [inline lib/Finance/Contract/Longcode.pm]: best to avoid `$a` and `$b` as variables

**Comment by junbon** [comment]
@binarytom Tick is handled differently as it has its own longcode string.

File: lib/Finance/Contract/Longcode.pm [hunk 3]
```diff
@@ -98,7 +98,18 @@ sub shortcode_to_longcode {
     if ($expiry_type eq 'intraday_fixed_expiry') {
         $when_end = [$date_expiry->datetime . ' GMT'];
     } elsif ($expiry_type eq 'intraday') {
-        $when_end = [Time::Duration::Concise->new(interval => $date_expiry->epoch - $date_start->epoch)->as_string];
+        my $interval_string = Time::Duration::Concise->new(interval => $date_expiry->epoch - $date_start->epoch)->as_string;
+        my @strings;
+        my @durations;
+        foreach my $interval ('hour', 'minute', 'second') {
+            if ($interval_string =~ /(\d+)\s$b/) {
```
Comment (basiliscos) [inline lib/Finance/Contract/Longcode.pm]: What is `$b` on the end of regex?

**Comment by basiliscos** [comment]
@junbon As the module `Time::Duration::Concise` belongs to binary.com, instead of converting interval to human-readable form, and then parse it, consider add method like

> Time::Duration::Concise->new(interval => $date_expiry->epoch - $date_start->epoch)->as_intervals.

which will return 4-values list like ($d, $h, $m, $s).

**Pull request merged**